db/migrate/20210801104200_create_storage_hosts_wwpn_candidates.rb

```diff
@@ -0,0 +1,14 @@
+class CreateStorageHostsWwpnCandidates < ActiveRecord::Migration[6.0]
+  def change
+    create_table :storage_hosts_wwpn_candidates do |t|
+      # stores WWPN candidate
+      t.string :candidates
```

Comment:
Is this a string holding an array?  Having a table, I'd expect one candidate per row.

Can you describe the relationships a little more in the OP?

Reply:
Will this contain all of the addresses that are not already in use?

Reply:
1.It will be one candidate per row. The name should be : 'candidate' . I will fix this, (this is mentioned by comment..)
2. This table will contain all of the addresses that are not already in use

Reply:
Should this column be singular `candidate` then?

Reply:
Please correct me if I'm wrong (I don't have much experience with FiberChannel) but my understanding is that a WWPN would be something like `10:00:00:00:c9:b0:51:32`.
Are you planning to store trillions of rows per EMS (minus the few we discover from refresh) in this table?
Are we going to seed the table?

Reply:
we are going to store only WWPN 'candidates'. These are WWPN which are not assigned yet
usually it is maximum 512 per node. But this is theoretically, since these WWPN are usually published and assigned one by one, and after that they 'disappear', so the actual amount of these 'candidates'  will be dozens per storage at the best.

Reply:
> Should this column be singular `candidate` then?

done